src/compile.rs

```diff
@@ -342,7 +352,11 @@ impl Call {
 /// The fold `(fold f)_n : E^(<2^n) × A → A`
 /// takes the list of type `E^(<2^n)` and an initial accumulator of type `A`,
 /// and it produces the final accumulator of type `A`.
-fn list_fold(bound: NonZeroPow2Usize, f: &ProgNode) -> Result<ProgNode, simplicity::types::Error> {
+fn list_fold(
+    ctx: &simplicity::types::Context,
+    bound: NonZeroPow2Usize,
+    f: &ProgNode,
+) -> Result<ProgNode, simplicity::types::Error> {
```

Comment:
In 7368e4597e17d9528c5bff43746c02135ce38cc6:

Here and elsewhere you should be able to take the ctx out of `f` so you don't need an extra parameter and you won't risk calling the method incorrectly.

Reply:
Great catch. Thanks for pointing this out.